lib/ansible/plugins/inventory/scaleway.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: scaleway
+    plugin_type: inventory
+    authors:
+      - Remy Leone <rleone@online.net>
+    short_description: Scaleway inventory source
+    description:
+        - Get inventory hosts from Scaleway
+    options:
+        regions:
+            description: Filter results on a specific Scaleway region
+            type: list
+            default:
+                - ams1
+                - par1
+        tags:
+            description: Filter results on a specific tag
+            type: list
+'''
+
+EXAMPLES = '''
+# scaleway_inventory.yml file in YAML format
+# Example command line: ansible-inventory --list -i scaleway_inventory.yml
+
+plugin: scaleway
+regions:
+  - ams1
+  - par1
+tags:
+  - foobar
+'''
+
+import json
+import os
+
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.module_utils.scaleway import SCALEWAY_LOCATION
+from ansible.module_utils.urls import open_url
+
+
+def _fetch_information(token, url):
+    try:
+        response = open_url(url,
+                            headers={'X-Auth-Token': token,
+                                     'Content-type': 'application/json'})
+    except Exception:
+        raise AnsibleError("Error while fetching %s" % url)
+
+    try:
+        raw_json = json.loads(response.read())
+    except ValueError:
+        raise AnsibleError("Incorrect JSON payload")
+
+    try:
+        return raw_json["servers"]
+    except KeyError:
+        raise AnsibleError("Incorrect format from the Scaleway API response")
+
+
+def _build_server_url(api_endpoint):
+    return "/".join([api_endpoint, "servers"])
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'scaleway'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        self.token = os.environ["SCW_TOKEN"]
```

Comment:
It looks like scaleway modules allow a series of fallback environment variables as well as a module option. If you wanted to do the same thing for this plugin you could do
```
options:
    ...
    oauth_token:
        env:
            # in order of precedence
            - name: SCW_TOKEN
            - name: SCW_API_KEY
            - name: SCW_OAUTH_TOKEN
```
Then you can use self.get_option('oauth_token').

Reply:
Done. Do you have other reviews?